### DISCUSSION TITLE
Can't testing when wrap my component in Controller exported from react-hook-form

### DISCUSSION THREAD

**Discussion by Keony1** [react-hook-form/react-hook-form#2449] [Q&A]
I dont know if it's just a doubt but, when i have one input component wrapped inside a Controller component of react-hook-form,
i cant realize a simple test.

just to clarify what is going on:

this is my test
![image](https://user-images.githubusercontent.com/30664147/89109541-79716680-d418-11ea-9a07-a3d986497900.png)

this is my component
![image](https://user-images.githubusercontent.com/30664147/89109561-a32a8d80-d418-11ea-9f68-068bc0c5405b.png)

and this is the error
![image](https://user-images.githubusercontent.com/30664147/89109569-b2114000-d418-11ea-860c-98fab5ceda81.png)

So, when i simply return a common input from my component, the test pass.

**Comment by bluebill1049** [comment]
oh don't mock `react-hook-form` let it re-render as it is, do as an integration test. it's much easier.

**Reply by bluebill1049** [reply]
this section may help: https://react-hook-form.com/advanced-usage#TestingForm

**Comment by Keony1** [comment]
I dont get it @bluebill1049 
in documentation examples a entire form are tested. In my case, i just want to test a input.

there is another way more easy to do that?

thank you!

**Reply by bluebill1049** [reply]
that's the idea, don't test individual input. test the entire form :)

**Reply by Keony1** [reply]
so i'm trying to test the entire form an the error persist:

![image](https://user-images.githubusercontent.com/30664147/89126158-e0932780-d4b9-11ea-87ff-647529891960.png)

**Reply by bluebill1049** [reply]
not sure why then, if this component can render properly in the browser, you should be able to test it.
